openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -276,7 +280,10 @@ def verify(self, user_id):
         (not expired) and bound to the given user.
         """
         try:
-            unsigned_data = signing.loads(self.signature, salt=self.key_salt, max_age=settings.SESSION_COOKIE_AGE)
+            unsigned_data = signing.TimestampSigner(
+                key=None, salt=self.key_salt, algorithm=settings.DEFAULT_HASHING_ALGORITHM
+            ).unsign_object(self.signature, serializer=signing.JSONSerializer, max_age=settings.SESSION_COOKIE_AGE)
```

Comment:
Checking old implementation in [django](https://github.com/django/django/blob/254df3a3bbc2f4df51f9e2070ab2e214553f67d0/django/core/signing.py#L170). Doing this way maintains the old behaviour as it is. I tried your suggestion but it breaks few tests in this [file](https://github.com/openedx/edx-platform/blob/7da99848a7d6436221b171ce8365daba57993675/openedx/core/djangoapps/safe_sessions/tests/test_safe_cookie_data.py#L20) because the are comparing `SafeCookieData` dict values and need further fixes in tests.

Reply:
I experimented with moving it to `__init__` and didn't see any test failures, but it's probably not worth digging into.